### DISCUSSION TITLE
PID control for TRV

### DISCUSSION THREAD

**Discussion by MartiX79** [ScratMan/HASmartThermostat#181] [Q&A]
Hi, at the moment I'm using HASmartThermostat PID controller to control my oil boiler which is controlled by a smart plug (on/off).
I also got TRV and am wondering how I may use the HASmartThermostat PID to control the TRV. How should I configure the config file?
what should be in sensor and target settings?

**Comment by Chupaka** [comment]
What is this TRV for?..

**Reply by MartiX79** [reply]
It's for the radiator, to open/close the radiator's valve. I'll have TRV on each radiator to control the temperature in each room.

**Reply by Chupaka** [reply]
Then the target_sensor should be a thermometer in the room, and target_temp is the temperature you expect there

**Reply by MartiX79** [reply]
But the TRV is climate type by itself. so, I'm not sure how to control it by PID.

**Reply by Chupaka** [reply]
As I can see, there are different TRVs around, so possible workflow depends on what you can control. Is there manual on/off mode for your TRV? Do you have additional thermometers in your rooms?

**Reply by MartiX79** [reply]
I'm using this one (link below). I'm planing to add extra temperature sensors. but the TRV have temp sensor too.
As I can see, there is no option to on/off the TRV, only regulate the temperature.

[trv - Aliexpress](https://www.aliexpress.com/item/1005005975848237.html?spm=a2g0o.productlist.main.13.426239ebE9soCH&algo_pvid=1ac127fc-8702-4644-9925-1aee4c8222dc&algo_exp_id=1ac127fc-8702-4644-9925-1aee4c8222dc-6&pdp_npi=4%40dis%21EUR%2196.63%2129.95%21%21%21745.06%21%21%402101c72a17000498522745869e5168%2112000035148666351%21sea%21IE%21702626488%21&curPageLogUid=zK6vIrjanMn3)

**Reply by Chupaka** [reply]
So, if you're going to use TRV temp sensor, then it will just work, you don't need Smart Thermostat.

If you add extra sensors - just look at TRV and "external" sensor difference and then you can adjust TRV target temp by Smart Thermostat to compensate this difference. Or leave it as is if the difference is minor.
